thumbor/handlers/__init__.py

```diff
@@ -250,16 +250,13 @@ def _load_results(self, context):
 
         return results, content_type
 
+    @gen.coroutine
     def _process_result_from_storage(self, result):
         if self.context.config.SEND_IF_MODIFIED_LAST_MODIFIED_HEADERS:
             # Handle If-Modified-Since & Last-Modified header
             try:
-                result_last_modified = self.context.modules.result_storage.last_updated()
-            except NotImplementedError:
-                logger.warn('last_updated method is not supported by your result storage service, hence If-Modified-Since & '
-                            'Last-Updated headers support is disabled.')
+                result_last_modified = yield gen.maybe_future(self.context.modules.result_storage.last_updated())
 
-            if result_last_modified:
```

Comment:
Removing this if might make 
`if result_last_modified <= date_modified_since` succeed/ fail even if it should not.

if `result_last_modified` is `None` then if clause is always true.